sqlmesh/core/schema_loader.py

```diff
@@ -76,5 +81,12 @@ def _get_columns(table: str) -> t.Optional[t.Dict[str, t.Any]]:
             if columns
         ]
 
+        # dont clobber existing entries from other gateways
+        entries_to_keep = (
+            [e for e in yaml.load(path) if e.get("gateway", None) != gateway]
```

Comment:
Wouldn't this result in duplicated entries for folks with existing external models?

Reply:
I'll add another test to verify the behaviour.

The intention was that the `gateway: `-specific entry would override the entry with no gateway specified when the file is loaded. So both can exist in the file without conflict, and one supersedes the other on load.

Does that seem like reasonable behaviour?

Reply:
Yeah, I'm more worried, that when users with existing files run the command to refresh their schemas, that'd see a huge delta cause we'd just insert all the entires all over again while also keeping the old ones.

Reply:
So it depends on whether users are explicitly setting `--gateway` or not. I observed that unless `--gateway` is explicitly set, it defaults to `None`.

There is no top-level fallback that sets it to the default gateway and then passes that to the loading / saving code. It seems that any code dealing with `gateway` must decide how to handle `None` vs a specific value.

So, a user wont see any duplicate entries if they usually don't set `--gateway` at all, because it will default to `None` and all the entries will be written without a gateway (which is the current behaviour).

If the user _does_ set `--gateway`, then yes the entries will be duplicated. However, this wont pose a problem on load because any gateway-specific entries will override entries with no gateway defined.

It will pose a problem on a `git diff` however. Since the user is using SQLMesh to manage their external models file anyway, I think it's reasonable to ask them to delete and re-create it. This should be minimal effort and will result in one set of external model definitions going forward.

I'm open to other suggestions though! I also thought about writing the gateway-specific models to their own file in the `external_models/` directory so that we wouldn't have to mutate an existing file. However, there is a potential for the file name to clash with a user-defined file

Reply:
I think as long as the original behavior is preserved if the user doesn't explicitly provide a gateway, I'm fine with that.